src/features/metadata-based-view/IconWithTooltip.js

```diff
@@ -0,0 +1,51 @@
+// @flow strict
+
+import React, { type Element } from 'react';
+import { injectIntl } from 'react-intl';
+
+import PlainButton from '../../components/plain-button';
+import Tooltip from '../../components/tooltip';
+import IconCheck from '../../icons/general/IconCheck';
+import IconClose from '../../icons/general/IconClose';
+import IconPencil from '../../icons/general/IconPencil';
+
+import { bdlGray62 } from '../../styles/variables';
+
+import { CANCEL_ICON_TYPE, EDIT_ICON_TYPE, SAVE_ICON_TYPE } from './constants';
+
+type IconType = typeof CANCEL_ICON_TYPE | typeof EDIT_ICON_TYPE | typeof SAVE_ICON_TYPE;
+type Props = {
+    className?: string,
+    onClick: void => void,
+    tooltipText: String,
+    type?: IconType,
```

Comment:
Instead of type type prop, what if we just rendered the `children` to keep the `IconWithTooltip` as reusable as possible? That way, you can pass whatever icon you want